src/type/definition.js

```diff
@@ -478,12 +478,12 @@ export type GraphQLIsTypeOfFn = (
   info: GraphQLResolveInfo
 ) => boolean
 
-export type GraphQLFieldResolveFn<TSource> = (
+export type GraphQLFieldResolveFn<TSource, TResult> = (
   source: TSource,
   args: {[argName: string]: mixed},
   context: mixed,
   info: GraphQLResolveInfo
-) => mixed
+) => TResult
```

Comment:
I'm curious if this actually aids in Flow's type checking? What issue would Flow not catch that it does catch after this change? I'd love to see an example so I can understand better.